integration-tests/spring-extension-support-test/src/main/java/com/example/application/autogrid/PropertyStringFilterSpecification.java

```diff
@@ -0,0 +1,199 @@
+ */
+public class PropertyStringFilterSpecification<T> {
+
+    private final PropertyStringFilter filter;
+    private final Class<?> javaType;
+
+    public PropertyStringFilterSpecification(PropertyStringFilter filter, Class<?> javaType) {
+        this.filter = filter;
+        this.javaType = javaType;
+    }
+
+    public Predicate toPredicate(Root<T> root, CriteriaQuery<?> query, CriteriaBuilder criteriaBuilder) {
+        String value = filter.getFilterValue();
+        Path<String> propertyPath = getPath(filter.getPropertyId(), root);
+        if (javaType == String.class) {
+            Expression<String> expr = criteriaBuilder.lower(propertyPath);
+            switch (filter.getMatcher()) {
+                case EQUALS:
+                    return criteriaBuilder.equal(expr, value.toLowerCase());
+                case CONTAINS:
+                    return criteriaBuilder.like(expr, "%" + value.toLowerCase() + "%");
+                case GREATER_THAN:
+                    throw new IllegalArgumentException("A string cannot be filtered using greater than");
+                case LESS_THAN:
+                    throw new IllegalArgumentException("A string cannot be filtered using less than");
+                default:
+                    break;
+            }
+
+        } else if (isNumber(javaType)) {
+            switch (filter.getMatcher()) {
+                case EQUALS:
+                    return criteriaBuilder.equal(propertyPath, value);
+                case CONTAINS:
+                    throw new IllegalArgumentException("A number cannot be filtered using contains");
+                case GREATER_THAN:
+                    return criteriaBuilder.greaterThan(propertyPath, value);
+                case LESS_THAN:
+                    return criteriaBuilder.lessThan(propertyPath, value);
+                default:
+                    break;
+            }
+        } else if (isBoolean(javaType)) {
+            Boolean booleanValue = Boolean.valueOf(value);
+            switch (filter.getMatcher()) {
+                case EQUALS:
+                    return criteriaBuilder.equal(propertyPath, booleanValue);
+                case CONTAINS:
+                    throw new IllegalArgumentException("A boolean cannot be filtered using contains");
+                case GREATER_THAN:
+                    throw new IllegalArgumentException("A boolean cannot be filtered using greater than");
+                case LESS_THAN:
+                    throw new IllegalArgumentException("A boolean cannot be filtered using less than");
+                default:
+                    break;
+            }
+        } else if (isDate(javaType)) {
+            var path = root.<Date>get(filter.getPropertyId());
+            var dateValue = Date.from(
+                    LocalDate.parse(value).atStartOfDay(ZoneId.systemDefault()).toInstant());
```

Comment:
Is it correct to use systemDefault here, even if copied? Because this could change if the system is configured differently, which would change the actual query to the database? Isn't UTC better here ?

Reply:
Good point. I will check

Reply:
I'll address this on another PR